Source/p_saveg.c

```diff
@@ -162,6 +162,50 @@ static void saveg_writep(const void *p)
 #define saveg_read_enum saveg_read32
 #define saveg_write_enum saveg_write32
 
+// [crispy] enumerate all thinker pointers
+static int P_ThinkerToIndex(thinker_t* thinker)
```

Comment:
Isn't this redundant with code that is already in P_ArchiveSpecials() or P_ArchiveThinkers()?

Reply:
I think we could remove `P_ThinkerToIndex` and convert the pointer to index in `P_ArchiveSpecials`. But we still need `P_IndexToPointer`.

Reply:
Like this:
```
if (th->function == T_Pusher)
{
  pusher_t *pusher = (pusher_t *) th;
  mobj_t *source = pusher->source;

  saveg_write8(tc_pusher);
  // convert to index
  pusher->source = i;
  saveg_write_pusher_t(pusher);
  // restore pointer
  pusher->source = source;
  continue;
}
```

Reply:
Let's please try to integrate this into the existing savegame code as closely as possible.

Reply:
It turns out my previous patch is not working.
To be honest, I like the current version better. The existing savegame code has separate functions for archiving thinkers and specials, so I think it would be ugly.